spackbot/pr_run_pipeline.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import logging
+import random
+import requests
+import re
+import os
+
+from .helpers import found, alias_regex
+from gidgethub import routing
+
+
+logger = logging.getLogger(__name__)
+router = routing.Router()
+
+# We can only make the request with a GITLAB TOKEN
+GITLAB_TOKEN = os.environ.get("GITLAB_TOKEN")
+
+
+async def run_pipeline(event, gh):
+    """
+    Make a request to re-run a pipeline.
+    """
+    # Get the pull request number
+    number = event.data["issue"]["pull_request"]["url"].split("/")[-1]
+
+    # We need the pull request branch
+    response = requests.get(event.data["issue"]["pull_request"]["url"])
```

Comment:
Seems like `event.data["issue"]["pull_request"]` could just be pulled out initially and used in both of these expressions.

Reply:
sure thing.